### ISSUE TITLE
[Improvement] LockManager.java comparison using reference equality instead of value equality

### ISSUE THREAD

**Issue by justinmclean** [datastrato/gravitino#2166]
### What would you like to be improved?

In LockManager.java:230: warning: [ReferenceEquality] Comparison using reference equality instead of value equality
      if (identifier == ROOT) {

See https://errorprone.info/bugpattern/ReferenceEquality

### How should we improve?

Does it mean 'if (Objects.equals(identifier, ROOT)) {' or 'if (identifier.equals(ROOT)) {'?

**Comment by YxAc** [comment]
In most cases, it is better to use the _Objects.equals()_ method. Because the _Objects.equals()_ method is safer when dealing with null values, it can correctly handle cases where both parameters are null without throwing a NullPointerException.

In the LockManager class, logically speaking, _identifier_ should not be null, so I think _identifier.equals(ROOT)_ is sufficient.

What do you think? Can I fix it with this idea? @justinmclean

**Comment by justinmclean** [comment]
Either is an improvement over the current code; I trust your judgment as to which is better.

**Comment by yuqi1129** [comment]
@justinmclean @YxAc 
I'm afraid we need to think twice about it, if we just use `equals` and the users create a name identifier that likes '/', the code may get an unexpected result.

**Comment by YxAc** [comment]
> @justinmclean @YxAc I'm afraid we need to think twice about it, if we just use `equals` and the users create a name identifier that likes '/', the code may get an unexpected result.

@yuqi1129 , I didn't get what you mean. Can you offer more details? thx

**Comment by yuqi1129** [comment]
> Can you offer more details? thx

What if a user creates a metalake with the name "/", obviously, it equals to the `ROOT`, but it's not the `ROOT`.

**Comment by YxAc** [comment]
> > Can you offer more details? thx
> 
> What if a user creates a metalake with the name "/", obviously, it equals to the `ROOT`, but it's not the `ROOT`.

if a user creates a metalake with the name "/", we expect return the root TreeLock? isn't it?
if using equals, return the root TreeLock exactly, isn't it our expected result? @yuqi1129 
i think we expected there is only one root treeLock in most cases.

**Comment by yuqi1129** [comment]
> we expect return the root TreeLock?
No, it's not the ROOT node. ROOT stands for the real root of the tree. Currently, we haven't banned users from naming metalake as '/'.

**Comment by YxAc** [comment]
> > we expect return the root TreeLock?
> 
> No, it's not the ROOT node. ROOT stands for the real root of the tree. Currently, we haven't banned users from naming metalake as '/'.

if we use _==_, it indeed not return the ROOT node, but i thought we should return the ROOT node, otherwise it's confused here.

if _==(reference equality)_ is deliberately used here to distinguish whether it is the real ROOT node, so this problem can be turned off? @yuqi1129

**Comment by yuqi1129** [comment]
> if _==(reference equality)_ is deliberately used here to distinguish whether it is the real ROOT node, so this problem can be turned off?

Yes, I use `==` (reference equality) deliberately, but any improvement and optimization is welcome, maybe we can come up with a better solution here.

**Comment by justinmclean** [comment]
If there was a good reason for using reference equality and it was deliberately done, then a comment should be added stating why this was done.
